chainerx_cc/chainerx/routines/math.cc

```diff
@@ -525,6 +525,69 @@ Array IfGreaterElse(const Array& x1, Scalar x2, Scalar pos, const Array& neg) {
 
 }  // namespace
 
+namespace {
+
+Array IfGreaterElse(const Array& x1, const Array& x2, const Array& pos, const Array& neg) {
+    CheckEqual(x1.dtype(), x2.dtype());
+    CheckEqual(x1.shape(), x2.shape());
+    Array out = EmptyLike(x1, x1.device());
+    {
+        NoBackpropModeScope scope{};
+        x1.device().IfGreaterElseAAAA(x1, x2, pos, neg, out);
+    }
+
+    {
+        BackwardBuilder bb{"if_greater_else", {pos, neg}, out};
+        if (BackwardBuilder::Target bt = bb.CreateTarget(0)) {
+            bt.Define([x1 = x1.AsGradStopped(), x2](BackwardContext& bctx) {
+                const Array& gout = *bctx.output_grad();
+                bctx.input_grad() = IfGreaterElse(x1, x2, gout, ZerosLike(gout, gout.device()));
+            });
+        }
+        if (BackwardBuilder::Target bt = bb.CreateTarget(1)) {
+            bt.Define([x1, x2 = x2.AsGradStopped()](BackwardContext& bctx) {
+                const Array& gout = *bctx.output_grad();
+                bctx.input_grad() = IfGreaterElse(x1, x2, ZerosLike(gout, gout.device()), gout);
+            });
+        }
+        bb.Finalize();
+
+        return out;
+    }
+}
+
+} // namespace
+
+namespace {
+
+void MinimumImpl(const Array& x1, const Array& x2, const Array& out) {
+    CheckEqual(x1.dtype(), x2.dtype());
+    CheckEqual(x1.shape(), x2.shape());
+    {
+        NoBackpropModeScope scope{};
+        x1.device().IfGreaterElseAAAA(x1, x2, x2, x1, out);
+    }
+
+    {
+        BackwardBuilder bb{"if_greater_else", {x2, x1}, out};
```

Comment:
This backward implementation seems to duplicate with `IfGreaterElse`'s one. How about defining `void IfGreaterElseImpl(const Array& x1, const Array& x2, const Array& pos, const Array& neg, const Array& out)` and just calling `IfGreaterElseImpl(x1, x2, x2, x1, out)` in this function?

Reply:
That makes sense. Thanks for the input @asi1024 .I'll work on that.

Reply:
Could `IfGreaterElse` be removed then?